crates/ruff_python_parser/src/lexer/ftstring.rs

```diff
@@ -4,31 +4,40 @@ use super::TokenFlags;
 
 /// The context representing the current f-string that the lexer is in.
 #[derive(Clone, Debug)]
-pub(crate) struct FStringContext {
+pub(crate) struct FTStringContext {
+    kind: FTStringKind,
     flags: TokenFlags,
 
-    /// The level of nesting for the lexer when it entered the current f-string.
+    /// The level of nesting for the lexer when it entered the current f/t-string.
     /// The nesting level includes all kinds of parentheses i.e., round, square,
     /// and curly.
     nesting: u32,
 
-    /// The current depth of format spec for the current f-string. This is because
+    /// The current depth of format spec for the current f/t-string. This is because
     /// there can be multiple format specs nested for the same f-string.
     /// For example, `{a:{b:{c}}}` has 3 format specs.
     format_spec_depth: u32,
 }
 
-impl FStringContext {
-    pub(crate) const fn new(flags: TokenFlags, nesting: u32) -> Self {
-        assert!(flags.is_f_string());
+impl FTStringContext {
+    pub(crate) const fn new(kind: FTStringKind, flags: TokenFlags, nesting: u32) -> Self {
+        match kind {
+            FTStringKind::FString => assert!(flags.is_f_string()),
+            FTStringKind::TString => assert!(flags.is_t_string()),
+        }
 
         Self {
+            kind,
             flags,
             nesting,
             format_spec_depth: 0,
         }
     }
 
+    pub(crate) const fn kind(&self) -> FTStringKind {
+        self.kind
+    }
```

Comment:
This seems redundant given that we also have `TokenFlags`. I haven't seen the usage bits yet but I'd suggest to:

* Remove `FTStringKind` entirely
* Keep the method but remove the field. The method returns `Kind::FString` if `flags.is_f_string` is true and t-string otherwise